trigger skedJobTrigger on sked__Job__c (after insert,after update) 
{
    try {
        if (trigger.isAfter) {
            if (trigger.isInsert) skedJobTriggerHandler.onAfterInsert(trigger.new);
            if (trigger.isUpdate) skedJobTriggerHandler.onAfterUpdate(trigger.new, trigger.oldMap);
        }
    } catch(DmlException e) 
    {
        System.debug('The following exception has occurred: ' + e.getMessage());
    }
}

//Apex Classes(Handler)

public class skedJobTriggerHandler {
    Public Static Set<Id> contactIds = new Set<Id>();
    public static void onAfterInsert(List<sked__job__c> jobList) {
        for(sked__job__c job : jobList) {
            if(job.sked__Job_Status__c == 'complete' && !String.isBlank(job.sked__Contact__c))
            {
                contactIds.add(job.sked__Contact__c);
            }
        }
        onCalloutResponse(contactIds);
    }
    public static void onAfterUpdate(List<sked__job__c> newJobValues, Map<id, sked__job__c> map_id_old) 
    {
        for( sked__Job__c job : newJobValues)
        {
            if(job.sked__Job_Status__c != map_id_old.get(job.Id).sked__Job_Status__c && job.sked__Job_Status__c == 'Complete' && !String.isBlank(job.sked__Contact__c))
            {
                conIds.add(job.sked__Contact__c);
            }
        }
        onCalloutResponse(contactIds);
    } 
    
    @future (callout=true) 
    public static void onCalloutResponse(Set<Id> contactIds) 
    {
        try{
            List<Contact> conDetailList = [SELECT id, Phone,LastName FROM Contact WHERE Id IN : conIds LIMIT 1];
            if(conDetailList.size()>0){
                String phone = conDetailList[0].Phone;
                String countryCode = 'US';
                String message = 'working on the exercise';
                sked.ApiResult.Sms response = sked.SkeduloAPI.sms(phone, countryCode, message );
                if (response.isSuccess()) {
                    System.debug('Message sent--->'+response); 
                } else {
                    System.debug('Error: ' + response.getError());
                } 
            }
        }
        
        catch(Exception e) 
        {
            System.debug('The following exception has occurred: ' + e.getMessage());
        }
        
    }
}
